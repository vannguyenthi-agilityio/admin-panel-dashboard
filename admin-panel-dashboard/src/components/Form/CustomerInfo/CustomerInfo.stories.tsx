import type { Meta, StoryObj } from "@storybook/react";

// Libs
import { useForm } from "react-hook-form";

// Components
import CustomerInfo from ".";

// Types
import { ICustomerData } from "@/types";

const meta: Meta<typeof CustomerInfo> = {
  title: "Components/Form/CustomerForm",
  component: CustomerInfo,
  tags: ["autodocs"],
  argTypes: {
    control: {
      description:
        "This is control from useForm of react-hook-form with type ProductBasicInfo({productName: string, description: string, providerName: string, weight: string, category: string, quantity: string})",
    }
  },
};

export default meta;

type Story = StoryObj<typeof meta>;

const CustomerInfoProvider = () => {
  const {
    control,
  } = useForm<ICustomerData>({
    defaultValues: {
      firstName: "Andrew",
      lastName: "Bachmaga",
      idNumber: 1234567890,
      email: "jone@gmail.com",
      dateOfBirth: { months: "", date: 0, years: 0 },
      phoneNumber: "20 7123 4567"
    },
  });

  return <CustomerInfo control={control}  />;
};

export const Default: Story = {
  render: () => <CustomerInfoProvider />,
};
