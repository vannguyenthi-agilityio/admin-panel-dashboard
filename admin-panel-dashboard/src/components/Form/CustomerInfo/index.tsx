import { memo } from "react";
import { Control, Controller } from "react-hook-form";

// Constants
import {
  REGEX,
  MESSAGES_ERROR,
  MESSAGES_WARNING,
  PREFIX_PHONE_NUMBER,
  FORM_TYPE
} from "@/constants";

// Components
import {
  Input,
  BirthDayField,
} from "@/components";

// Types
import { ICustomerData } from "@/types";

// Helper
import { validateBirthDate, formatPhoneNumber } from "@/helpers";

// Icons
import { UKIcon } from "@/components/Icons";

interface ICustomerInfo {
  control: Control<ICustomerData>;
  type?: FORM_TYPE;
}

const CustomerInfo = ({ control, type = FORM_TYPE.CREATE }: ICustomerInfo) => {
  const isDisable = type === FORM_TYPE.DETAIL;
  return (
    <div className="grid grid-cols-1 items-start bg-white rounded-lg shadow-box-icon-default auto-rows-min h-full">
      <Controller
        name="firstName"
        control={control}
        rules={{
          required: MESSAGES_ERROR.FIELD_REQUIRED,
          minLength: {
            value: 2,
            message: MESSAGES_ERROR.MIN_LENGTH_2,
          },
          pattern: {
            value: REGEX.NAME,
            message: MESSAGES_ERROR.NAME_INVALID,
          },
        }}
        render={({
          field: { value, onChange, onBlur, ...rest },
          fieldState: { error },
        }) => (
          <div className="min-h-[90px] w-full">
            <Input
              type="text"
              id="first-name"
              data-testid="first-name"
              label="First Name"
              required
              disabled={isDisable}
              errorMessage={error?.message}
              value={value}
              onChange={onChange}
              onBlur={onBlur}
              {...rest }
            />
          </div>
        )}
      />

      <Controller
        name="lastName"
        control={control}
        rules={{
          required: MESSAGES_ERROR.FIELD_REQUIRED,
          minLength: {
            value: 2,
            message: MESSAGES_ERROR.MIN_LENGTH_2,
          },
          pattern: {
            value: REGEX.NAME,
            message: MESSAGES_ERROR.NAME_INVALID,
          },
        }}
        render={({ field: { value, onChange, onBlur, ...rest }, fieldState: { error } }) => (
          <div className="min-h-[90px] w-full">
            <Input
              type="text"
              id="last-name"
              data-testid="last-name"
              label="Last Name"
              required
              disabled={isDisable}
              errorMessage={error?.message}
              value={value}
              onChange={onChange}
              onBlur={onBlur}
              {...rest}
            />
          </div>
        )}
      />

      <Controller
        name="idNumber"
        control={control}
        rules={{
          required: MESSAGES_ERROR.FIELD_REQUIRED,

          pattern: {
            value: REGEX.NUMBER,
            message: MESSAGES_ERROR.ID_NUMBER_INVALID,
          },
        }}
        render={({ field: { value, onChange, onBlur, ...rest }, fieldState: { error } }) => {
          return(
            <div className="min-h-[90px] w-full">
              <Input
                id="id-number"
                data-testid="id-number"
                label="ID Number"
                required
                disabled={isDisable}
                errorMessage={error?.message}
                value={value}
                onChange={onChange}
                onBlur={onBlur}
                {...rest}
              />
              {!error?.message && (
                <p className="pt-1 text-xs text-seldom">
                  {MESSAGES_WARNING.ID_NUMBER}
                </p>
              )}
            </div>
          )
        }}
      />
      <Controller
          control={control}
          name="dateOfBirth"
          rules={{
            validate: validateBirthDate,
          }}
          render={({ field: { value, onChange }, fieldState: { error } }) => {
            return (
            <div className="w-full min-h-[112px] mt-4 mb-5">
              <BirthDayField
                required
                disabled={isDisable}
                value={value}
                onChange={onChange}
                errorMessage={error?.message}
              />
            </div>)
          }}
        />
      <Controller
        control={control}
        rules={{
          pattern: {
            value: REGEX.PHONE_NUMBER,
            message: MESSAGES_ERROR.INVALID_PHONE_NUMBER,
          },
        }}
        name="phoneNumber"
        render={({ field: { value, onChange, onBlur, ...rest }, fieldState: { error } }) => {
          const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
            const valuePhone = event.target.value;
            const phoneValueFormatter = formatPhoneNumber(valuePhone);
            onChange(phoneValueFormatter);
          };

          return (
            <div className="min-h-[90px] col-span-1">
              <Input
                type="text"
                contentPrefix={PREFIX_PHONE_NUMBER}
                className="pl-[60px]"
                label="Phone number"
                id="phoneNumber"
                leftElement={<UKIcon className="h-4 w-4" />}
                value={value}
                disabled={isDisable}
                errorMessage={error?.message}
                onChange={handleChange}
                onBlur={onBlur}
                {...rest}
              />
            </div>
          );
        }}
      />
      <Controller
        control={control}
        rules={{
          required: MESSAGES_ERROR.FIELD_REQUIRED,
          pattern: {
            value: REGEX.EMAIL,
            message: MESSAGES_ERROR.EMAIL_INVALID,
          },
        }}
        name="email"
        render={({ field: { value, onChange, onBlur, ...rest }, fieldState: { error } }) => (
          <div className="min-h-[90px] w-full">
            <Input
              label="Email"
              id="email"
              required
              disabled={isDisable}
              value={value}
              errorMessage={error?.message}
              onChange={onChange}
              onBlur={onBlur}
              {...rest}
            />
          </div>
        )}
      />
    </div>
  )
};

export default memo(CustomerInfo);
