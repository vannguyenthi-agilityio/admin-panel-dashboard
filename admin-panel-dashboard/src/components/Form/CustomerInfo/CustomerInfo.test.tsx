// Libs
import { useForm } from "react-hook-form";
import { render } from "@testing-library/react";

// Components
import CustomerInfo from ".";

// Types
import { ICustomerData } from "@/types";

const mockControllerNonError = (name: string, rule: Object) => ({
  field: {
    name: name,
    rules: rule,
    onChange: jest.fn(),
    value: [],
  },
  formState: { errors: {} },
  fieldState: { error: {} },
});

jest.mock("react-hook-form", () => ({
  ...jest.requireActual("react-hook-form"),
  // Mock the useForm hook
  useForm: jest.fn(() => {
    return {
      formState: {
        errors: {},
      },
    };
  }),
  // Mock the Controller component
  Controller: (props: any) =>
    props.render(mockControllerNonError(props.name, props.rule)),
}));

describe("UserInfo Component", () => {
  const { control } = useForm<ICustomerData>();

  const renderCustomerInfoComponent = () =>
    render(
      <CustomerInfo
        control={control}
      />,
    );

  it("should match snapshot", () => {
    const { container } = renderCustomerInfoComponent();
    expect(container).toMatchSnapshot();
  });
});
