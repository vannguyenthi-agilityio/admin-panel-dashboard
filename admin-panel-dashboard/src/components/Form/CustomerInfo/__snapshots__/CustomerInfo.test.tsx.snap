// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo Component should match snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-1 items-start bg-white rounded-lg shadow-box-icon-default auto-rows-min h-full"
  >
    <div
      class="min-h-[90px] w-full"
    >
      <label
        class="text-sm font-medium after:content-['*'] after:text-red-500 after:text-md"
        for="first-name"
      >
        First Name
      </label>
      <div
        class="relative w-full sm:w-auto min-w-[320px]"
      >
        <input
          class="peer block w-full flex-1 font-primary font-light appearance-none rounded-lg border-gray-light bg-white px-2 text-sm text-tertiary placeholder-text-gray focus:outline-none hover:enabled:border-dark-gray aria-invalid:!border-red disabled:text-gray disabled:cursor-not-allowed border placeholder:opacity-0 focus:placeholder:opacity-100 py-[9px]"
          data-testid="first-name"
          id="first-name"
          name="firstName"
          placeholder=" "
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="min-h-[90px] w-full"
    >
      <label
        class="text-sm font-medium after:content-['*'] after:text-red-500 after:text-md"
        for="last-name"
      >
        Last Name
      </label>
      <div
        class="relative w-full sm:w-auto min-w-[320px]"
      >
        <input
          class="peer block w-full flex-1 font-primary font-light appearance-none rounded-lg border-gray-light bg-white px-2 text-sm text-tertiary placeholder-text-gray focus:outline-none hover:enabled:border-dark-gray aria-invalid:!border-red disabled:text-gray disabled:cursor-not-allowed border placeholder:opacity-0 focus:placeholder:opacity-100 py-[9px]"
          data-testid="last-name"
          id="last-name"
          name="lastName"
          placeholder=" "
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="min-h-[90px] w-full"
    >
      <label
        class="text-sm font-medium after:content-['*'] after:text-red-500 after:text-md"
        for="id-number"
      >
        ID Number
      </label>
      <div
        class="relative w-full sm:w-auto min-w-[320px]"
      >
        <input
          class="peer block w-full flex-1 font-primary font-light appearance-none rounded-lg border-gray-light bg-white px-2 text-sm text-tertiary placeholder-text-gray focus:outline-none hover:enabled:border-dark-gray aria-invalid:!border-red disabled:text-gray disabled:cursor-not-allowed border placeholder:opacity-0 focus:placeholder:opacity-100 py-[9px]"
          data-testid="id-number"
          id="id-number"
          name="idNumber"
          placeholder=" "
          value=""
        />
      </div>
      <p
        class="pt-1 text-xs text-seldom"
      >
        The ID number should be 10 characters
      </p>
    </div>
    <div
      class="w-full min-h-[112px] mt-4 mb-5"
    >
      <p
        class="text-2xl text-tertiary mb-2"
      >
        Date of Birth
      </p>
      <div
        class="flex flex-col md:flex-row gap-6"
      >
        <div
          class="w-full"
        >
          <label
            class="text-grey text-sm font-medium capitalize after:content-['*'] after:text-red-500 after:text-md"
          >
            Month
          </label>
          <div
            class="relative"
          >
            <select
              class="w-full font-primary font-light placeholder:text-slate-400 border border-slate-200 rounded-lg transition duration-300 ease focus:outline-none focus:border-slate-400 hover:border-slate-400 shadow-sm focus:shadow-md appearance-none bg-transparent text-dark h-10 text-sm px-2 py-2 cursor-pointer"
              data-testid="month"
            >
              <option
                class="cursor-pointer"
                data-testid="option-0"
                value="0"
              >
                January
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1"
                value="1"
              >
                February
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2"
                value="2"
              >
                March
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-3"
                value="3"
              >
                April
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-4"
                value="4"
              >
                May
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-5"
                value="5"
              >
                June
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-6"
                value="6"
              >
                July
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-7"
                value="7"
              >
                August
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-8"
                value="8"
              >
                September
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-9"
                value="9"
              >
                October
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-10"
                value="10"
              >
                November
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-11"
                value="11"
              >
                December
              </option>
            </select>
            <div
              class="ml-1 absolute top-[18px] right-3 text-slate-700"
            >
              <svg
                fill="none"
                height="5"
                viewBox="0 0 8 5"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.25 0.75L4 4.25L0.75 0.75"
                  stroke="#adb5bd"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="w-full"
        >
          <label
            class="text-grey text-sm font-medium capitalize after:content-['*'] after:text-red-500 after:text-md"
          >
            Date
          </label>
          <div
            class="relative"
          >
            <select
              class="w-full font-primary font-light placeholder:text-slate-400 border border-slate-200 rounded-lg transition duration-300 ease focus:outline-none focus:border-slate-400 hover:border-slate-400 shadow-sm focus:shadow-md appearance-none bg-transparent text-dark h-10 text-sm px-2 py-2 cursor-pointer"
              data-testid="date"
            />
            <div
              class="ml-1 absolute top-[18px] right-3 text-slate-700"
            >
              <svg
                fill="none"
                height="5"
                viewBox="0 0 8 5"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.25 0.75L4 4.25L0.75 0.75"
                  stroke="#adb5bd"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="w-full"
        >
          <label
            class="text-grey text-sm font-medium capitalize after:content-['*'] after:text-red-500 after:text-md"
          >
            Year
          </label>
          <div
            class="relative"
          >
            <select
              class="w-full font-primary font-light placeholder:text-slate-400 border border-slate-200 rounded-lg transition duration-300 ease focus:outline-none focus:border-slate-400 hover:border-slate-400 shadow-sm focus:shadow-md appearance-none bg-transparent text-dark h-10 text-sm px-2 py-2 cursor-pointer"
              data-testid="year"
            >
              <option
                class="cursor-pointer"
                data-testid="option-1900"
                value="1900"
              >
                1900
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1901"
                value="1901"
              >
                1901
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1902"
                value="1902"
              >
                1902
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1903"
                value="1903"
              >
                1903
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1904"
                value="1904"
              >
                1904
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1905"
                value="1905"
              >
                1905
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1906"
                value="1906"
              >
                1906
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1907"
                value="1907"
              >
                1907
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1908"
                value="1908"
              >
                1908
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1909"
                value="1909"
              >
                1909
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1910"
                value="1910"
              >
                1910
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1911"
                value="1911"
              >
                1911
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1912"
                value="1912"
              >
                1912
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1913"
                value="1913"
              >
                1913
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1914"
                value="1914"
              >
                1914
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1915"
                value="1915"
              >
                1915
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1916"
                value="1916"
              >
                1916
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1917"
                value="1917"
              >
                1917
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1918"
                value="1918"
              >
                1918
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1919"
                value="1919"
              >
                1919
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1920"
                value="1920"
              >
                1920
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1921"
                value="1921"
              >
                1921
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1922"
                value="1922"
              >
                1922
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1923"
                value="1923"
              >
                1923
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1924"
                value="1924"
              >
                1924
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1925"
                value="1925"
              >
                1925
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1926"
                value="1926"
              >
                1926
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1927"
                value="1927"
              >
                1927
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1928"
                value="1928"
              >
                1928
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1929"
                value="1929"
              >
                1929
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1930"
                value="1930"
              >
                1930
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1931"
                value="1931"
              >
                1931
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1932"
                value="1932"
              >
                1932
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1933"
                value="1933"
              >
                1933
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1934"
                value="1934"
              >
                1934
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1935"
                value="1935"
              >
                1935
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1936"
                value="1936"
              >
                1936
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1937"
                value="1937"
              >
                1937
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1938"
                value="1938"
              >
                1938
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1939"
                value="1939"
              >
                1939
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1940"
                value="1940"
              >
                1940
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1941"
                value="1941"
              >
                1941
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1942"
                value="1942"
              >
                1942
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1943"
                value="1943"
              >
                1943
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1944"
                value="1944"
              >
                1944
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1945"
                value="1945"
              >
                1945
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1946"
                value="1946"
              >
                1946
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1947"
                value="1947"
              >
                1947
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1948"
                value="1948"
              >
                1948
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1949"
                value="1949"
              >
                1949
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1950"
                value="1950"
              >
                1950
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1951"
                value="1951"
              >
                1951
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1952"
                value="1952"
              >
                1952
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1953"
                value="1953"
              >
                1953
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1954"
                value="1954"
              >
                1954
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1955"
                value="1955"
              >
                1955
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1956"
                value="1956"
              >
                1956
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1957"
                value="1957"
              >
                1957
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1958"
                value="1958"
              >
                1958
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1959"
                value="1959"
              >
                1959
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1960"
                value="1960"
              >
                1960
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1961"
                value="1961"
              >
                1961
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1962"
                value="1962"
              >
                1962
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1963"
                value="1963"
              >
                1963
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1964"
                value="1964"
              >
                1964
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1965"
                value="1965"
              >
                1965
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1966"
                value="1966"
              >
                1966
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1967"
                value="1967"
              >
                1967
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1968"
                value="1968"
              >
                1968
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1969"
                value="1969"
              >
                1969
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1970"
                value="1970"
              >
                1970
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1971"
                value="1971"
              >
                1971
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1972"
                value="1972"
              >
                1972
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1973"
                value="1973"
              >
                1973
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1974"
                value="1974"
              >
                1974
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1975"
                value="1975"
              >
                1975
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1976"
                value="1976"
              >
                1976
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1977"
                value="1977"
              >
                1977
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1978"
                value="1978"
              >
                1978
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1979"
                value="1979"
              >
                1979
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1980"
                value="1980"
              >
                1980
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1981"
                value="1981"
              >
                1981
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1982"
                value="1982"
              >
                1982
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1983"
                value="1983"
              >
                1983
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1984"
                value="1984"
              >
                1984
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1985"
                value="1985"
              >
                1985
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1986"
                value="1986"
              >
                1986
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1987"
                value="1987"
              >
                1987
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1988"
                value="1988"
              >
                1988
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1989"
                value="1989"
              >
                1989
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1990"
                value="1990"
              >
                1990
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1991"
                value="1991"
              >
                1991
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1992"
                value="1992"
              >
                1992
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1993"
                value="1993"
              >
                1993
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1994"
                value="1994"
              >
                1994
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1995"
                value="1995"
              >
                1995
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1996"
                value="1996"
              >
                1996
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1997"
                value="1997"
              >
                1997
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1998"
                value="1998"
              >
                1998
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-1999"
                value="1999"
              >
                1999
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2000"
                value="2000"
              >
                2000
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2001"
                value="2001"
              >
                2001
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2002"
                value="2002"
              >
                2002
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2003"
                value="2003"
              >
                2003
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2004"
                value="2004"
              >
                2004
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2005"
                value="2005"
              >
                2005
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2006"
                value="2006"
              >
                2006
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2007"
                value="2007"
              >
                2007
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2008"
                value="2008"
              >
                2008
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2009"
                value="2009"
              >
                2009
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2010"
                value="2010"
              >
                2010
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2011"
                value="2011"
              >
                2011
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2012"
                value="2012"
              >
                2012
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2013"
                value="2013"
              >
                2013
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2014"
                value="2014"
              >
                2014
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2015"
                value="2015"
              >
                2015
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2016"
                value="2016"
              >
                2016
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2017"
                value="2017"
              >
                2017
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2018"
                value="2018"
              >
                2018
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2019"
                value="2019"
              >
                2019
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2020"
                value="2020"
              >
                2020
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2021"
                value="2021"
              >
                2021
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2022"
                value="2022"
              >
                2022
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2023"
                value="2023"
              >
                2023
              </option>
              <option
                class="cursor-pointer"
                data-testid="option-2024"
                value="2024"
              >
                2024
              </option>
            </select>
            <div
              class="ml-1 absolute top-[18px] right-3 text-slate-700"
            >
              <svg
                fill="none"
                height="5"
                viewBox="0 0 8 5"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.25 0.75L4 4.25L0.75 0.75"
                  stroke="#adb5bd"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <p
        class="text-xs text-red-500 pt-1"
      >
        Age must be greater than 18
      </p>
    </div>
    <div
      class="min-h-[90px] col-span-1"
    >
      <label
        class="text-sm font-medium"
        for="phoneNumber"
      >
        Phone number
      </label>
      <div
        class="relative w-full sm:w-auto min-w-[320px]"
      >
        <div
          class="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3 text-dark-gray w-[40px]"
        >
          <svg
            class="h-4 w-4"
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_2239_2203)"
            >
              <path
                d="M19.6553 16.5193H0.345087C0.172522 16.5193 0.0325905 16.3794 0.0325905 16.2068V3.79308C0.0325905 3.62052 0.172522 3.48059 0.345087 3.48059H19.6554C19.8279 3.48059 19.9679 3.62052 19.9679 3.79308V16.2068C19.9678 16.3794 19.8279 16.5193 19.6553 16.5193Z"
                fill="#41479B"
                stroke="#6C757D"
                stroke-width="0.0646926"
              />
              <path
                d="M20.0002 3.79308C20.0002 3.60266 19.8458 3.44824 19.6553 3.44824H18.1126L11.7243 7.63366V3.44824H8.27603V7.63366L1.88781 3.44824H0.345087C0.154658 3.44824 0.000244141 3.60266 0.000244141 3.79308V4.68492L5.48104 8.2758H0.000244141V11.7241H5.48104L0.000244141 15.315V16.2068C0.000244141 16.3973 0.154658 16.5517 0.345087 16.5517H1.88785L8.27611 12.3663V16.5517H11.7244V12.3663L18.1126 16.5517H19.6554C19.8458 16.5517 20.0002 16.3973 20.0002 16.2068V15.315L14.5194 11.7241H20.0002V8.27584H14.5194L20.0002 4.68492V3.79308Z"
                fill="#F5F5F5"
              />
              <path
                d="M20.0002 8.96549H11.0347V3.44824H8.96572V8.96549H0.000244141V11.0344H8.96572V16.5517H11.0347V11.0344H20.0002V8.96549Z"
                fill="#FF4B55"
              />
              <path
                d="M6.99049 11.7236L0.0101318 16.2547C0.0345458 16.4213 0.17185 16.5512 0.34517 16.5512H0.820129L8.25721 11.7237H6.99049V11.7236Z"
                fill="#FF4B55"
              />
              <path
                d="M13.5312 11.7236H12.2645L19.6908 16.5441C19.8639 16.5256 20.0005 16.3843 20.0005 16.2063V15.9228L13.5312 11.7236Z"
                fill="#FF4B55"
              />
              <path
                d="M0.000549316 4.17332L6.32002 8.27541H7.58673L0.201447 3.48145C0.0834787 3.53625 0.000549316 3.65402 0.000549316 3.79265V4.17332Z"
                fill="#FF4B55"
              />
              <path
                d="M12.9914 8.2758L19.9885 3.73375C19.9594 3.57273 19.8251 3.44824 19.6557 3.44824H19.1616L11.7247 8.2758H12.9914Z"
                fill="#FF4B55"
              />
            </g>
            <defs>
              <clippath
                id="clip0_2239_2203"
              >
                <rect
                  fill="#fff"
                  height="20"
                  width="20"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <span
          class="absolute top-[12px] text-xs min-h-[30px] text-darker left-[35px]"
        >
          +44
        </span>
        <input
          class="peer block w-full flex-1 font-primary font-light appearance-none rounded-lg border-gray-light bg-white px-2 text-sm text-tertiary placeholder-text-gray focus:outline-none hover:enabled:border-dark-gray aria-invalid:!border-red disabled:text-gray disabled:cursor-not-allowed border placeholder:opacity-0 focus:placeholder:opacity-100 py-[9px] pl-[60px]"
          id="phoneNumber"
          name="phoneNumber"
          placeholder=" "
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="min-h-[90px] w-full"
    >
      <label
        class="text-sm font-medium after:content-['*'] after:text-red-500 after:text-md"
        for="email"
      >
        Email
      </label>
      <div
        class="relative w-full sm:w-auto min-w-[320px]"
      >
        <input
          class="peer block w-full flex-1 font-primary font-light appearance-none rounded-lg border-gray-light bg-white px-2 text-sm text-tertiary placeholder-text-gray focus:outline-none hover:enabled:border-dark-gray aria-invalid:!border-red disabled:text-gray disabled:cursor-not-allowed border placeholder:opacity-0 focus:placeholder:opacity-100 py-[9px]"
          id="email"
          name="email"
          placeholder=" "
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
